File: src/nodes/global_scope.zig
const nd = @import("node.zig");

pub const NodeGlobalScope = struct {
    statement: *nd.Node,

    pub fn init(statement: *nd.Node) NodeGlobalScope {
        return NodeGlobalScope{ .statement = statement };
    }

    pub fn get_statement(self: *const NodeGlobalScope) *nd.Node {
        return self.statement;
    }
};

File: src/nodes/global_scope.rs
use crate::nodes::node::Node;

pub struct NodeGlobalScope {
    statement: Node,
}

impl NodeGlobalScope {
    pub fn init(statement: Node) -> NodeGlobalScope {
        NodeGlobalScope { statement }
    }
    
    pub fn get_statement(&self) -> &Node {
        &self.statement
    }
}